sympy/polys/polytools.py

```diff
@@ -3569,7 +3569,7 @@ def real_roots(f, multiple=True, radicals=True):
         [CRootOf(x**3 + x + 1, 0)]
 
         """
-        reals = sympy.polys.rootoftools.CRootOf.real_roots(f, radicals=radicals)
+        reals = sorted(sympy.polys.rootoftools.CRootOf.real_roots(f, radicals=radicals))
```

Comment:
Can you add a direct test for this? What polynomial wasn't giving the roots in the right order?

Reply:
I believe after the original is simplified, it uses `Poly(-27*_x**4 - 90*_x**3 - 108*_x**2 - 54*_x - 9, _x, domain='ZZ')`.

I've been drilling down a bit to see if there's a more fundamental issue. Eventually, it gets to `polyutils._sort_factors()`. Isn't that else clause dead code?

https://github.com/sympy/sympy/blob/dd08a11b11a9cef066fb80b54a57f83887ccb852/sympy/polys/polyutils.py#L154-L166

Reply:
Never mind, it's not if multiple is set to False. Perhaps this should be the case when it's called...

Reply:
> Hypothesis found another example in master, `(Poly(x + 1)**2*Poly(-5*x - 1)).real_roots()` (which is also flaky). I think it's fixed here too. I am now running this on this branch to see if it finds anything. I'm guessing it won't, but I'll let you know if it turns anything up.

@asmeurer Did you want me to update this PR with the `is_disjoint` fix, or is it the `sorted` amendment that you're testing?

Reply:
Let me know if that direct test is what you had in mind.